packages/framework-core/src/services/graphql/graphql-query-generator.ts

```diff
@@ -141,75 +132,137 @@ export class GraphQLQueryGenerator {
-  }
+  private generateArrayFilterFor(property: PropertyMetadata): GraphQLInputObjectType {
+    const filterName = `${property.name}PropertyFilter`
 
-  private generateFilterFor(type: AnyClass): GraphQLInputObjectType {
-    const filterName = `${type.name}PropertyFilter`
     if (!this.generatedFiltersByTypeName[filterName]) {
-      const graphQLValueType = this.typeInformer.getGraphQLTypeFor(type)
+      const propFilters: GraphQLInputFieldConfigMap = {}
+      property.typeInfo.parameters.forEach((param) => {
+        const primitiveType = this.typeInformer.getPrimitiveExtendedType(param.type)
+        let graphqlType: GraphQLScalarType
+        switch (primitiveType) {
+          case Boolean:
+            graphqlType = GraphQLBoolean
+            break
+          case String:
+            graphqlType = GraphQLString
+            break
+          case Number:
+            graphqlType = GraphQLFloat
+            break
+          default:
+            graphqlType = GraphQLJSONObject
+            break
+        }
+        propFilters.includes = {
+          ...propFilters.includes,
+          type: graphqlType,
+        }
+      })
+
       this.generatedFiltersByTypeName[filterName] = new GraphQLInputObjectType({
         name: filterName,
-        fields: {
-          operation: { type: new GraphQLNonNull(this.operationEnumFor(type)) },
-          values: { type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(graphQLValueType))) },
-        },
+        fields: propFilters,
       })
     }
     return this.generatedFiltersByTypeName[filterName]
   }
 
-  private operationEnumFor(type: AnyClass): GraphQLEnumType {
-    const operationEnumName = `${type.name}Operations`
-    if (!this.generatedOperationEnumsByTypeName[operationEnumName]) {
-      this.generatedOperationEnumsByTypeName[operationEnumName] = new GraphQLEnumType({
-        name: operationEnumName,
-        values: this.generateOperationEnumValuesFor(type),
-      })
+  private generateFilterFor(prop: PropertyMetadata): GraphQLInputObjectType | GraphQLScalarType {
+    const filterName = `${prop.typeInfo.name}PropertyFilter`
+
+    if (prop.typeInfo.type === undefined) return GraphQLJSONObject
+
+    const primitiveType = this.typeInformer.getPrimitiveExtendedType(prop.typeInfo.type)
+    if (primitiveType === Array) return this.generateArrayFilterFor(prop)
+    if (!this.generatedFiltersByTypeName[filterName]) {
+      const graphQLPropType = this.typeInformer.getGraphQLTypeFor(primitiveType)
+      let fields: Thunk<GraphQLInputFieldConfigMap> = {}
+
+      if (!this.typeInformer.isPrimitiveType(graphQLPropType)) {
+        let nestedProperties: GraphQLInputFieldConfigMap = {}
+        const properties = getPropertiesMetadata(prop.typeInfo.type)
+        if (properties.length > 0) {
+          this.typeInformer.generateGraphQLTypeFromMetadata({ class: prop.typeInfo.type, properties })
```

Comment:
By calling this function the custom types like `Money` are created and included in the schema. Without this line, those properties would be `JSONObject`

Reply:
Aha! Thanks. Yep, that's because Money is not included among the types of the typeInformer.  I think the TypeInformer requires a little refactor to not require to pass all the types it can handle in the constructor, but to generate them as requested. But for another PR.
👏